src/tests/test_builtin_diffs.py

```diff
@@ -174,7 +174,7 @@ def test_unary_function(backend, func, y_d, domain):
     if isinstance(ret, da.Array):
         ret.compute()
 
-    assert_allclose(x.diff, y_d_arr)
+    assert_allclose(x.diff[ret.name].value, y_d_arr)
```

Comment:
What does `.name` represent here?

Reply:
You can think of it as the id of each `DiffArray`, and the name of each `DiffArray` is not equal.

Reply:
What is it based on, UUID?

Reply:
No, I use a counter to assign a `name` to each `DiffArray`.
```python
class DiffArray(np.ndarray):
    count = 0

    def __init__(self, arr):
        if isinstance(arr, DiffArray):
            self._value = arr.value
            self._name = arr.name
            self._parents = arr._parents
            self._vjp = arr._vjp
            self._diff = arr._diff
            self._jacobian = arr._jacobian

        from udiff import SKIP_SELF

        with SKIP_SELF:
            arr = np.asarray(arr)

        self._value = arr
        self._name = "var_{}".format(DiffArray.count)
        self._parents = []
        self._vjp = None
        self._diff = None
        self._jacobian = None

        DiffArray.count += 1
```

Reply:
This is once again, not thread safe. I suggest a `uuid.uuid4()`

Reply:
OK.